### PR TITLE
Recognize user in space header

### PR DESCRIPTION
**Pull request by RandomAshes**
## Issue
Connects #401 

## What was done
- [x] Added user cdsid to the dashboard account dropdown toggle
- [x] Refactored header and account dropdown

Screen Shot 2020-09-29 at 2 01 05 PM

## How to test
1. Login and go into a space. Ensure your cdsid shows up on the right side of the header next the the account dropdown.
2. Ensure the actions in the action dropdown work properly still.

NOTE: if you go to the flipping sweet board, there is no current user, so nothing will show.

### CODE REVIEW

**Review by LowBudgetMan — CHANGES_REQUESTED** [review]
pls

File: ui/src/Auth/TokenProvider.tsx [hunk 1]
```diff
@@ -16,8 +16,30 @@
...
+};
+
+export const getUserNameFromAccessToken = (): string => {
+    try {
+        const decodedAccessToken = getDecodedToken();
+        // @ts-ignore
+        const userName = decodedAccessToken.sub;
+        return userName;
+    } catch {
+        return '';
```
Comment (egriff38ford) [inline ui/src/Auth/TokenProvider.tsx:43]: Will we wind up with "Welcome, " if jwtdecoder doesn't work? Should we remove the comma if there's no name?

Comment (RandomAshes) [reply ui/src/Auth/TokenProvider.tsx:43]: There's a check in the html so it won't actually show up.

**Review by egriff38ford — APPROVED** [review]
🧊 🧊 👶

**Pull request merged**